lib/plugins/package/lib/packageService.js

```diff
@@ -70,12 +71,21 @@ module.exports = {
     const functionObject = this.serverless.service.getFunction(functionName);
     const funcPackageConfig = functionObject.package || {};
 
+    // use the artifact in function config if provided
+    if (funcPackageConfig.artifact) {
+      const filePath = path.join(this.serverless.config.servicePath, funcPackageConfig.artifact);
+      functionObject.package.artifact = filePath;
```

Comment:
I found out that this should be `functionObject.package.artifact` and not `functionObject.artifact`. I referred to the code below earlier. Changed that as well.